actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -118,7 +118,7 @@ def extract_parameterized_parts(route, options, recall)
             end
           end
 
-          parameterized_parts.keep_if { |_, v| v }
+          parameterized_parts.delete_if { |_, v| v.nil? }
```

Comment:
Could we use compact here?
```suggestion
          parameterized_parts.compact!
```

Reply:
Ah, `compact` works, but `compact!` does not. Thanks for the suggestion! either way it is nicer then `delete_if` :)

Reply:
Right, `compact!` only works if you leave the `parameterize_parts` line after (because it returns `nil` if nothing is removed). `compact` will always return the new value because it creates a copy

Reply:
I see! I suppose `delete_if` doesn't create a copy so maybe it would be better to use `compact!` and leave the `parameterized_parts` at the end. like you originally suggested.